terraform/environments/data-platform/iam.tf

```
@@ -99,27 +132,61 @@ data "aws_iam_policy_document" "athena_load_lambda_function_policy" {
   }
 }
 
-data "aws_iam_policy_document" "iam_policy_document_for_authorizer_lambda" {
+data "aws_iam_policy_document" "landing_to_raw_lambda_policy" {
+  source_policy_documents = [data.aws_iam_policy_document.log_to_bucket.json, data.aws_iam_policy_document.read_metadata.json]
+
   statement {
-    sid       = "LambdaLogGroup"
-    effect    = "Allow"
-    actions   = ["logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents"]
-    resources = ["arn:aws:logs:${local.region}:${local.account_id}:log-group:/aws/lambda/*"]
+    sid    = "AllowLambdaToCreateLogGroup"
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup"
+    ]
+    resources = [
+      format("arn:aws:logs:eu-west-2:%s:*", data.aws_caller_identity.current.account_id)
+    ]
   }
   statement {
-    sid    = "s3LogAccess"
+    sid    = "AllowLambdaToWriteLogsToGroup"
     effect = "Allow"
     actions = [
-      "s3:GetObject",
-      "s3:PutObject",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    resources = [
+      format("arn:aws:logs:eu-west-2:%s:*", data.aws_caller_identity.current.account_id)
     ]
+  }
+  statement {
+    sid     = "getLandingData"
+    effect  = "Allow"
+    actions = ["s3:GetObject*"]
     resources = [
-      "${module.s3-bucket.bucket.arn}/logs/*"
+      "${module.data_landing_s3_bucket.bucket.arn}/*"
     ]
   }
+  statement {
+    sid     = "putRawData"
+    effect  = "Allow"
+    actions = ["s3:PutObject*"]
+    resources = [
+      "${module.data_s3_bucket.bucket.arn}/raw/*"
```

```suggestion
    actions = [
        "s3:PutObject*",
        "s3:ListBucket",
    ]
    resources = [
      "${module.data_s3_bucket.bucket.arn}/raw/*",
      "${module.data_s3_bucket.bucket.arn}/raw"
```
As above, need to access the bucket as well as the 'key' path